fsspec/asyn.py

```diff
@@ -92,6 +92,50 @@ def get_loop():
     return loop[0]
 
 
+try:
+    import resource
+except ImportError:
+    resource = None
+    ResourceError = OSError
+else:
+    ResourceEror = resource.error
+
+_DEFAULT_SOFT_LIMIT = 1024
+
+
+def _get_soft_limit():
+    if resource is None:
+        return _DEFAULT_SOFT_LIMIT
+
+    try:
+        soft_limit, _ = resource.getrlimit(resource.RLIMIT_NOFILE)
```

Comment:
Docs imply this can be called `RLIMIT_OFILE` on BSD (including macos?)

Reply:
Aren't they are aliases (i am not on mac, so can't test :().

Reply:
I don't know. I just checked from mac, and it's NOFILE for osx too! I guess darwin is not real BSD, but maybe we need to worry about it still.

Reply:
Tested on freebsd, openbsd, netbest but can't find any `RLIMIT_OFILE`, weird hmm.
```
gcc220$ python3
Python 3.8.6 (default, Oct 13 2020, 09:04:17) 
[Clang 10.0.1 ] on openbsd6
Type "help", "copyright", "credits" or "license" for more information.
>>> import resource
>>> resource.RLIMIT_OFILE
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'resource' has no attribute 'RLIMIT_OFILE'
>>> resource.RLIMIT_NOFILE
8
```
```
gcc300$ python3.7
Python 3.7.7 (default, May 13 2020, 14:31:34) 
[GCC 7.4.0] on netbsd9
Type "help", "copyright", "credits" or "license" for more information.
>>> import resource
>>> resource.RLIMIT_OFILE
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'resource' has no attribute 'RLIMIT_OFILE'
>>> resource.RLIMIT_NOFILE
8
```
```
Python 3.6.7 (default, Feb  7 2019, 01:15:51) 
[GCC 4.2.1 Compatible FreeBSD Clang 6.0.1 (tags/RELEASE_601/final 335540)] on freebsd12
Type "help", "copyright", "credits" or "license" for more information.
>>> import resource
>>> resource.RLIMIT_OFILE
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'resource' has no attribute 'RLIMIT_OFILE'
>>> resource.RLIMIT_NOFILE
8
>>> resource.getrlimit(resource.RLIMIT_NOFILE)
(87363, 87363)
```

Reply:
That's a big limit! :) If only that were true everywhere, we would never have seen this problem. I think macos's default is 256, but you can configure it.

OK, we can assume, then, that this some relic.

Reply:
> That's a big limit! :)

Not so great for a public machine with 2 cores / 2 gb of ram TBH :) In my local system (ubuntu) soft limit is 1024, which is quite easy to hit.